examples/app/src/main/kotlin/io/provenance/scope/examples/app/SmartKeyHsm.kt

```diff
@@ -0,0 +1,132 @@
+package io.provenance.scope.examples.app
+
+import com.fortanix.sdkms.v1.ApiClient
+import com.fortanix.sdkms.v1.api.AuthenticationApi
+import com.fortanix.sdkms.v1.api.SignAndVerifyApi
+import com.fortanix.sdkms.v1.auth.ApiKeyAuth
+import io.grpc.ManagedChannelBuilder
+import io.provenance.scope.contract.annotations.Record
+import io.provenance.scope.contract.proto.Specifications.PartyType
+import io.provenance.scope.encryption.model.SmartKeyRef
+import io.provenance.scope.encryption.util.toJavaPublicKey
+import io.provenance.scope.examples.SimpleExample.ExampleName
+import io.provenance.scope.examples.app.utils.SingleTx
+import io.provenance.scope.examples.app.utils.TransactionService
+import io.provenance.scope.examples.app.utils.getScope
+import io.provenance.scope.examples.app.utils.persistBatchToProvenance
+import io.provenance.scope.examples.contract.SimpleExampleContract
+import io.provenance.scope.examples.contract.SimpleExampleScopeSpecification
+import io.provenance.scope.sdk.Affiliate
+import io.provenance.scope.sdk.Client
+import io.provenance.scope.sdk.ClientConfig
+import io.provenance.scope.sdk.SharedClient
+import io.provenance.scope.sdk.SignedResult
+import java.net.URI
+import java.util.UUID
+import java.util.concurrent.TimeUnit
+
+data class ExampleNameHydrateHsm(
+    @Record("name") val name: ExampleName,
+)
+
+fun main(args: Array<String>) {
+    println("Executing example with Smart Key HSM crypto operations!")
+
+    // Creates Provenance grpc client. This is used for fetching Provenance account information, as well as
+    // simulating and broadcasting TXs.
+    val provenanceUri = URI(System.getenv("PROVENANCE_GRPC_URL"))
+    val channel = ManagedChannelBuilder
+        .forAddress(provenanceUri.host, provenanceUri.port)
+        .also {
+            if (provenanceUri.scheme.endsWith("s")) {
+                it.useTransportSecurity()
+            } else {
+                it.usePlaintext()
+            }
+        }
+        .build()
+    val transactionService = TransactionService(System.getenv("CHAIN_ID"), channel)
+
+    // Creates a Smart Key API Client
+    val smartKeyApiKey = System.getenv("SMART_KEY_API_KEY")
+    val smartKeyClient = ApiClient().apply {
+        setBasicAuthString(smartKeyApiKey)
+        com.fortanix.sdkms.v1.Configuration.setDefaultApiClient(this)
+
+        // authenticate with API
+        val authResponse = AuthenticationApi(this).authorize()
+        val auth = getAuthentication("bearerToken") as ApiKeyAuth
+        auth.apiKey = authResponse.accessToken
+        auth.apiKeyPrefix = "Bearer"
+    }
+    val signAndVerifyApi = SignAndVerifyApi(smartKeyClient)
+
+    // Creates a P8e scope client that will be used for contract execution.
+    val publicKey = System.getenv("SMART_KEY_PUBLIC_KEY").toJavaPublicKey()
+    val smartKeyUuid = UUID.fromString(System.getenv("SMART_KEY_UUID"))
```

Comment:
is there some env file that we should put these two values in (`SMART_KEY_PUBLIC_KEY` and `SMART_KEY_UUID`)? Or should they all be supplied manually when running the example, along with the api key?

Reply:
Good call. I should explicitly document that. I left them out of the host.env because these would be specific to a third party's account. I don't think this example will ever get run unless they were already interested and setup to make use of smart key.

Reply:
sounds good